src/grouping-functions-and-node-capabilities.md

```diff
@@ -8,9 +8,9 @@ The group a JSON-RPC function belongs to is indicated by the `prefix_` in its na
 
 This JSON-RPC interface is intended to be implemented on various types of nodes:
 
-- Full nodes, which hold in their database all block headers and bodies, plus the storage of recent blocks.
+- Full nodes, which hold in their database all headers from the genesis block to the most recent block, plus the bodies and storage of the most recent blocks (typically the latest 256).
```

Comment:
Full nodes store the bodies of all blocks. The sentence was accurate, and the number of blocks that are kept is unstable and arbitrary and thus better left out.

```suggestion
- Full nodes, which hold in their database all block headers and bodies, plus the storage of recent blocks.
```

Reply:
Oh wow! I didn't know that! If that's the case, then I suppose that [the substrate docs](https://docs.substrate.io/learn/networks-and-nodes/#full-nodes) should also be updated, because they clearly state that:

> By default, full nodes are configured to store only the most recent 256 blocks and to discard state older than that—with the exception of the genesis block—to prevent the full node from growing indefinitely and consuming all available disk space. You can configure the number of blocks a full node retains.

> Although older blocks are discarded, full nodes retain all of the [block headers](https://docs.substrate.io/reference/glossary/#header) from the genesis block to the most recent block to validate that the state is correct.

Reply:
Actually, the next sentence already hints that full-nodes also store the bodies:
 
> Because the full node has access to all of the block headers, it can be used to rebuild the state of the entire blockchain by executing all of the blocks from the genesis block. Thus it requires much more computation to retrieve information about some previous state, and an archive should generally be used instead.

If they are able to rebuild the state, that also implies that they have the bodies (transactions) and not just the headers. So, yeah, I think that those docs are a bit misleading. Thanks @tomaka !

Reply:
I accepted @tomaka 's suggestion so that this PR is about adding the missing description only as originally created.

Reply:
Wait! I have substrate developers telling me that:

> Block bodies should also be discarded depending on the --blocks-pruning setting of the node.

Reply:
![image](https://github.com/paritytech/json-rpc-interface-spec/assets/8620144/5218daa9-265f-4ab6-a09c-7a4811464b62)

Reply:
Ok, it's good to know that I wasn't the only one under that wrong impression! So, yeah, we should probably improve the substrate docs. 🙂